spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/core/QueueMessageChannel.java

```diff
@@ -111,6 +122,9 @@ private SendMessageRequest prepareSendMessageRequest(Message<?> message) {
 
 		Map<String, MessageAttributeValue> messageAttributes = getMessageAttributes(
 				message);
+		if (!(message.getPayload() instanceof  String)) {
+			messageAttributes.put(CONTENT_TYPE, new MessageAttributeValue().withStringValue(message.getPayload().getClass().getName()));
```

Comment:
why the `contenttype` will be the classname? I think it should be `application/json` but I think this should be provided by the one who produces the message.

Reply:
I was wondering what if someone calls QueueMessagingTemplate. 	public Message<?> receive. Should message returned for ex. be like  Message<TestPerson> or should it be Message<String> where String represents Json of object? That is reason why I was using class name, was not sure if Receive method should do the casting itself. With applcation/json I would like to document or alwys add MappingJackson2MessageConverter to Converters in QueueMessagingTemplate. Since without MappingJackson2MessageConverter it could possible fail to convert if user is to add converter which cannot convert from json, so return value will alwys be null. Currently it is only being added if there is no MessageConverter being added in creation of QueueMessagingTemplate. In new draft i added test that will fail so you can check it out. What do you think is better approach?